samples/subsys/rtio/producer_consumer/src/main.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/rtio/rtio.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
+
+/* Our producer is a timer (interrupt) but could be a thread as well */
+struct producer {
+	struct mpsc io_q;
+};
+
+/* Our producer function, could be done in a thread or interrupt, here
+ * we are producing cycle count values from a timer interrupt.
+ */
+static void producer_periodic(struct k_timer *timer)
+{
+	struct producer *p = timer->user_data;
+
+	LOG_INF("producer %p trigger", (void *)p);
+
+	struct mpsc_node *n = mpsc_pop(&p->io_q);
+
+	if (n == NULL) {
+		LOG_WRN("producer overflowed consumers");
+	}
+
+	struct rtio_iodev_sqe *iodev_sqe = CONTAINER_OF(n, struct rtio_iodev_sqe, q);
+
+	/* Only accept read/rx requests */
+	if (iodev_sqe->sqe.op != RTIO_OP_RX) {
+		rtio_iodev_sqe_err(iodev_sqe, -EINVAL);
+		return;
+	}
+
+	/* Get the rx buffer with a minimum/maximum size pair to fill with the current time */
+	uint8_t *buf = NULL;
+	uint32_t buf_len = 0;
+	int rc = rtio_sqe_rx_buf(iodev_sqe, sizeof(uint64_t), sizeof(uint64_t), &buf, &buf_len);
+
+	LOG_INF("buf %p, len %u", (void *)buf, buf_len);
+
+	if (rc < 0) {
+		/* buffer wasn't available or too small */
+		rtio_iodev_sqe_err(iodev_sqe, -ENOMEM);
+		return;
+	}
+
+	/* We now have a buffer we can produce data into and then signal back to the consumer */
+	uint64_t cycle_count = k_cycle_get_64();
+
+	/* "Produce" a timestamp */
+	memcpy(buf, (uint8_t *)&cycle_count, sizeof(uint64_t));
```

Comment:
Suggestion: Simplify this memcpy by having `uint64_t *cycle_count` returned directly from `rtio_sqe_rx_buf()`.

I get having `buf` and `buf_len` improves clarify, so alternative we could just cast it to an `uint64_t *`?